src/components/views/messages/MPollBody.tsx

```diff
@@ -317,14 +309,61 @@ export default class MPollBody extends React.Component<IBodyProps, IState> {
                 },
             );
         });
+    }
+
+    private selectOption(answerId: string) {
+        if (this.isEnded()) {
+            return;
+        }
+        const userVotes = this.collectUserVotes();
+        const userId = this.context.getUserId();
+        const myVote = userVotes.get(userId)?.answers[0];
+        if (answerId === myVote) {
+            return;
+        }
 
+        const response = PollResponseEvent.from([answerId], this.props.mxEvent.getId()).serialize();
+        this.sendContextEvent(response);
         this.setState({ selected: answerId });
     }
 
     private onOptionSelected = (e: React.FormEvent<HTMLInputElement>): void => {
         this.selectOption(e.currentTarget.value);
     };
 
+    // allow multiple selections
+    private selectMultipleOptions = (answerId: string) => {
+        if (this.isEnded()) {
+            return;
+        }
+        const userVotes = this.collectUserVotes();
+        const userId = this.context.getUserId();
+        const myVotes = userVotes.get(userId)?.answers ?? [];
+        const poll = this.props.mxEvent.unstableExtensibleEvent as PollStartEvent;
+
+        if (!myVotes.includes(answerId)) {
+            if (myVotes.length === poll.maxSelections) {
```

Comment:
To be safe I would suggest

```suggestion
            if (myVotes.length >= poll.maxSelections) {
```